### DISCUSSION TITLE
Update an item without loading it first

### DISCUSSION THREAD

**Discussion by m-urubek** [pnp/pnpcore#1140] [Q&A] [answered]
From what I understand, updating an IListItem can be done only using method IListItem.Update() (and its variants). But to perform this operation I need to have an instance of the IListItem which can be created only by requesting the item from db by LoadListDataAsStream or GetItemById. If I understand this correctly, it is done this way because PnP does version checks and possibly some other background operations. But for some specifics of my application, I don't need to do any version checks and need to have the update as fast as possible so I want to perform just one simple db update operation without loading the item first. Is there any way to do that?

**Comment by jansenbe** [comment]
@m-urubek : currently this is not possible natively in PnP Core SDK, you can however write your logic with raw API calls (see https://pnp.github.io/pnpcore/using-the-sdk/basics-customapirequests.html#making-a-custom-spo-rest-request). To do an update you have to have the unique id or URI of the item to update. For example we use internally `_api/web/lists/getbyid(guid'{Parent.Id}')/items({Id})/ValidateUpdateListItem`, so you're required to know the id of the item to update and id of the list containing the item

**Comment by m-urubek** [comment, marked as answer]
I have figured it out. Following method works. Took me a few hours of going through source code.

```
public static void Update(PnPContext context, string listName, int itemId) {
        Type? listItemType = Type.GetType("PnP.Core.Model.SharePoint.ListItem, PnP.Core");
        IListItem listItemInstance = (IListItem)listItemType.GetConstructors().FirstOrDefault().Invoke(null);

        PropertyInfo? idProperty = listItemType.GetProperties().Where(pr => pr.Name == "Id").FirstOrDefault();
        idProperty.SetValue(listItemInstance, itemId);

        listItemInstance.Values.Add("Title", "test1");

        IList list = context.Web.Lists.GetByTitle(listName);
        listItemInstance.Parent = list;

        Type? baseDataModelType = Type.GetType("PnP.Core.Model.SharePoint.BaseDataModel, PnP.Core");
        PropertyInfo? metadataProperty = listItemType.GetProperties().Where(pr => pr.Name == "Metadata").FirstOrDefault();
        Dictionary<string, string> metadataInstance = (Dictionary<string, string>)metadataProperty.GetValue(listItemInstance);
        metadataInstance.Add("restId", itemId.ToString());
        metadataProperty.SetValue(listItemInstance, metadataInstance);

        listItemInstance.PnPContext = context;

        listItemInstance.Update();
    }
```

**Reply by jansenbe** [reply]
Cool, nice to see you've managed to reverse engineer this and don't have to resort to building the REST call yourselves.
